### ISSUE TITLE
Unsupported left operand type for Callable in Union

### ISSUE THREAD

**Issue by jolaf** [python/mypy#13075]
The following code:
```python3
from collections.abc import Callable
from typing import Any

A = Callable[[Any], Any]
B = Callable[[Any, Any], Any]
C = A | B
```
produces the following output:
```shell
$ mypy test.py
test.py:6:5: error: Unsupported left operand type for | ("object")  [operator]
    C = A | B
        ^
Found 1 error in 1 file (checked 1 source file)
```
This looks like a false positive to me.
```shell
$ mypy --version
mypy 0.961 (compiled: yes)

$ python3 --version
Python 3.10.4

$ lsb_release -d
Description:	Ubuntu 22.04 LTS
```

**Comment by AlexWaygood** [comment]
Duplicate of #12393
